src/main/java/com/example/imagescaler/upscale/configuration/ApiProperties.java

```diff
@@ -0,0 +1,13 @@
+package com.example.imagescaler.upscale.configuration;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+@ConfigurationProperties(prefix = "api")
+@Getter
+@Setter
+public class ApiProperties {
```

Comment:
I would rename this to `UpscaleApiConfig` with an inner class `Configuration`.

The `UpscaleApiConfig` would have the annotation `@EnableConfigurationProperties(UpscaleApiConfig.Configuration.class)`.

The `UpscaleApiConfig.Configuration` would have the `@ConfigurationProperties(prefix="upscale-api")` annotation.

You can change `@Getter` + `@Setter` with `@Data`.